### PR TITLE
Fix homepage RTL layout

### LINKED ISSUE #165: Layout: Front page - RTL
There are some minor adjustments that should be made for RTL viewing on the homepage.

| Desktop | Tablet | Mobile |
|-|-|-|
| localhost_8888_(Desktop) | localhost_8888_(iPad) | localhost_8888_(Samsung Galaxy S20 Ultra) |

### PR DESCRIPTION
**Pull request by adamwoodnz**
Fixes #165 

Several things are addressed in this PR:

1. Vertical alignment of the columns
2. Overflow of the mobile sub nav
3. Consistent vertical spacing within each section

## How to test

1. Load the homepage
2. Check normal LTR responsive layout
3. Add attribute `dir="rtl"` to the root page element
4. Check RTL responsive layout
5. Use the mobile dropdown sub nav, and ensure buttons and input text are laid out correctly, with no overflow in both directions

#### LTR

| Before | After |
|-|-|
| localhost_8888_(Desktop) before ltr | localhost_8888_(Desktop) after ltr |

NOTE: The [designs](https://www.figma.com/file/2WxlJFzMJvqPfZL1EkAOVp/Developer-Resources?node-id=2014%3A17011&t=rfauehWRGoOWR6Cs-1) show the right 'API Reference' column horizontally centered on large screens, but not on a regular desktop. For simplicity I've opted to keep it centered all the time.

#### RTL

| Before | After |
|-|-|
| localhost_8888_(Desktop) | localhost_8888_(Desktop) after rtl |

### Mobile

| LTR Before | LTR After | RTL Before | RTL After |
|-|-|-|-|
| localhost_8888_(Samsung Galaxy S20 Ultra) before ltr | localhost_8888_(Samsung Galaxy S20 Ultra) after ltr | localhost_8888_(Samsung Galaxy S20 Ultra) before rtl | localhost_8888_(Samsung Galaxy S20 Ultra) after rtl |

### CODE REVIEW

File: source/wp-content/themes/wporg-developer-2023/patterns/front-page-content.php [hunk 1]
```diff
@@ -24,8 +24,8 @@
...
-<div class="wp-block-group has-white-color has-charcoal-2-background-color has-text-color has-background has-link-color has-aquamarine-background-color" style="padding-right:var(--wp--preset--spacing--edge-space);padding-left:var(--wp--preset--spacing--edge-space)"><!-- wp:group {"align":"wide","style":{"spacing":{"padding":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|60"}}},"layout":{"type":"default"}} -->
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"right":"var:preset|spacing|edge-space","left":"var:preset|spacing|edge-space"}}},"backgroundColor":"aquamarine","className":"has-white-color has-charcoal-2-background-color has-text-color has-background has-link-color","layout":{"type":"constrained"}} -->
+<div class="wp-block-group alignfull has-white-color has-charcoal-2-background-color has-text-color has-background has-link-color has-aquamarine-background-color" style="padding-right:var(--wp--preset--spacing--edge-space);padding-left:var(--wp--preset--spacing--edge-space)"><!-- wp:group {"align":"wide","style":{"spacing":{"padding":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|60"}}},"layout":{"type":"default"}} -->
```
Comment (adamwoodnz) [inline source/wp-content/themes/wporg-developer-2023/patterns/front-page-content.php:28]: Without these full alignment settings the content is constrained in the editor, maybe this is due to a Gutenberg update

Screen Shot 2023-02-20 at 4 26 39 PM

File: source/wp-content/themes/wporg-developer-2023/src/style/style.scss [hunk 2]
```diff
@@ -120,20 +134,16 @@ body[class] {
...
+	.home-section-docs-api {
+		.wp-block-column {
+			padding-left: var(--wp--preset--spacing--edge-space) !important;
+			padding-right: var(--wp--preset--spacing--edge-space) !important;
+
+			&[style*="border-right-"] {
+				border-bottom: 1px solid var(--wp--preset--color--light-grey-1);
+				border-right: none !important;
+			}
+		}
```
Comment (adamwoodnz) [inline source/wp-content/themes/wporg-developer-2023/src/style/style.scss:146]: I'm unsure whether this pattern is used in other pages, so maybe the styles shouldn't be removed.

For this particular section, once the horizontal padding is removed from the columns to allow the layout to work in both directions, and added to the group wrapper instead, the border styling breaks, as it doesn't stretch edge to edge anymore.

IMO this changeset is simpler (no negative margin) and less brittle, but it's also less reusable. Open to discussion.

File: source/wp-content/themes/wporg-developer-2023/src/style/style.scss [hunk 3]
```diff
@@ -120,20 +134,19 @@ body[class] {
...
 @media (max-width: 781px) {
-	.wp-block-column[style*="border-right-"] {
-		border-bottom: 1px solid var(--wp--preset--color--light-grey-1);
-		border-right: none !important;
-	}
+	.home-section-docs-api {
+		padding-left: 0 !important;
+		padding-right: 0 !important;
 
-	.wp-block-column[style*="border-right-"][style*="padding-right"] {
```
Comment (adamwoodnz) [inline source/wp-content/themes/wporg-developer-2023/src/style/style.scss:103]: I'm unsure whether this pattern is used in other pages, so maybe the styles shouldn't be removed.

For this particular section (Documentation & API Reference), once the horizontal padding is removed from the columns to allow the layout to work in both directions, the border styling breaks.

IMO this changeset is simpler (no negative margin) and less brittle, but it's also less reusable. Open to discussion.

**Review by StevenDufresne — APPROVED** [review]
I'm getting a border-right.

Screenshot 2023-03-27 at 3 02 19 PM

Other than that, looks good.

**Comment by ryelle** [comment]
I ran into a border issue when adding the RTL support in main: https://github.com/WordPress/wporg-main-2022/pull/126 Maybe that fix should be added to the parent theme? (not sure if that's relevant to this issue, just sharing based on the comment above)

**Comment by adamwoodnz** [comment]
> I'm getting a border-right.

Not sure why that was showing, it was supposed to be `linen` to match the background. It's unnecessary anyway, fixed in https://github.com/WordPress/wporg-developer/pull/222/commits/9d8451c5f7948a1641379d7c9334a43ae3baab97

**Comment by ryelle** [comment]
> Not sure why that was showing, it was supposed to be linen to match the background. It's unnecessary anyway, fixed in https://github.com/WordPress/wporg-developer/commit/9d8451c5f7948a1641379d7c9334a43ae3baab97

I remembered what caused this— it's [a bug in the RTL styles generated from GB](https://github.com/WordPress/gutenberg/pull/44170), where the border style is set for the block's opposite side, causing the browser to render a default border.

In any case, removing the border works 👍🏻

**Pull request merged**